diff --git a/addressfield_phone.module b/addressfield_phone.module
index e17bbfb..d4a0131 100644
--- a/addressfield_phone.module
+++ b/addressfield_phone.module
@@ -16,8 +16,11 @@ function addressfield_phone_ctools_plugin_directory($module, $plugin) {
 function addressfield_phone_field_attach_presave($entity_type, $entity) {
   $addressfields = addressfield_phone_addressfield_types();
   foreach ($addressfields as $field_name) {
-    if (isset($entity->$field_name)){
-      $default_fields = array_keys(addressfield_default_values());
+    if (isset($entity->$field_name)) {
+      $field = field_info_field($field_name);
+      list($id, $vid, $bundle) = entity_extract_ids($entity_type, $entity);
+      $instance = field_info_instance($entity_type, $field_name, $bundle);
+      $default_fields = array_keys(addressfield_default_values($field, $instance));
       $default_fields[] = 'element_key';
       foreach ($entity->$field_name as $lang_code => $fields) {
         foreach ($fields as $delta => $fieldset) {
